cmd/network-console-collector/resources/openshift/prometheus.yaml

```diff
@@ -0,0 +1,207 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: network-console-prometheus
+  labels:
+    app.kubernetes.io/part-of: network-console
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - services
+  - configmaps
+  - pods
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - delete
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  - statefulsets
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - delete
+- apiGroups:
+  - apps
+  resources:
+  - daemonsets
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - networking.k8s.io
+  resources:
+  - ingresses
+  verbs:
+  - get
+  - list
+  - watch
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: network-console-prometheus
+  labels:
+    app.kubernetes.io/part-of: network-console
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: network-console-prometheus
+  labels:
+    app.kubernetes.io/part-of: network-console
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: network-console-prometheus
+subjects:
+- kind: ServiceAccount
+  name: network-console-prometheus
+---
+apiVersion: v1
+data:
+  prometheus.yml: |
+    global:
+      scrape_interval:     15s
+      evaluation_interval: 15s
+    alerting:
+      alertmanagers:
+        - static_configs:
+            - targets:
+    rule_files:
+    scrape_configs:
+      - job_name: network-console-service
```

Comment:
minor detail: in the original prometheus scrape configuration we just had a static target for the service name and used tls_config.insecure_skip_verify. This version uses service discovery instead because:
1. TLS: openshift service certs are signed for the fully qualified name (e.g. mysvc.namespace.svc.cluster.local instead of mysvc.) Getting the target from the kube api lets us get that fully qualified name without templating the yaml.
2. Include labels from the service resource similar to the default prometheus community helm chart setup - which may be nice if any users want to use federate or remote write metrics from this instance to somewhere central.